### PR TITLE
[docs] Update FSDP instructions and add DeepSpeed evaluate/predict example

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Updates FSDP instructions based on the actual behaviour of the code. I realised that we incorrectly ask the user to instantiate layers within the context manager (the intended usage I think, however for use cases of already existing models this isn't preferable it seems), and have updated the docs to reflect the appropriate behaviour (just wrap layers in the context).

I also added an evaluation example for DeepSpeed!

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/advanced/advanced_gpu.rst [hunk 1]
```diff
@@ -359,6 +363,24 @@ Also please have a look at our :ref:`deepspeed-zero-stage-3-tips` which contains
...
+.. code-block:: python
+
+    from pytorch_lightning import Trainer
+    from pytorch_lightning.plugins import DeepSpeedPlugin
+    from deepspeed.ops.adam import FusedAdam
+
+
+    class MyModel(pl.LightningModule):
+        ...
+
```
[the remark this answers is not in the document]
Comment (SeanNaren) [reply docs/source/advanced/advanced_gpu.rst:377]: Don't think we need this for showing inference?

**Review by tchaton — APPROVED** [review]

File: docs/source/advanced/advanced_gpu.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaushikb11) [reply docs/source/advanced/advanced_gpu.rst:377]: I mentioned it because you imported it.
It's not mandatory!

**Review by Borda — APPROVED** [review]

File: docs/source/advanced/advanced_gpu.rst [hunk 2]
```diff
@@ -129,22 +127,28 @@ Below is an example of using both ``wrap`` and ``auto_wrap`` to create your mode
...
+            self.block = nn.Sequential(nn.Linear(32, 32), nn.ReLU())
+            self.final_block = nn.Sequential(nn.Linear(32, 32), nn.ReLU())
 
         def configure_sharded_model(self):
-            # Created within sharded model context, modules are instantly sharded across processes
-            # as soon as they are wrapped with ``wrap`` or ``auto_wrap``
+            # modules are sharded across processes
+            # as soon as they are wrapped with ``wrap`` or ``auto_wrap``.
+            # During the forward/backward passes, weights synced across processes
+            # and de-allocated once computation is complete, saving memory.
```
Comment (awaelchli) [inline docs/source/advanced/advanced_gpu.rst:140]:
```suggestion
            # During the forward/backward passes, weights get synced across processes
            # and de-allocated once computation is complete, saving memory.
```
a verb was missing, right?

File: docs/source/advanced/advanced_gpu.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (SeanNaren) [reply docs/source/advanced/advanced_gpu.rst:377]: ohh my bad! let me remove the import

File: docs/source/advanced/advanced_gpu.rst [hunk 3]
```diff
@@ -359,6 +363,24 @@ Also please have a look at our :ref:`deepspeed-zero-stage-3-tips` which contains
     trainer.predict()
 
 
+You can also use the Lightning Trainer to run predict or evaluate with DeepSpeed once the model has been trained.
+
+.. code-block:: python
+
+    from pytorch_lightning import Trainer
+    from pytorch_lightning.plugins import DeepSpeedPlugin
+    from deepspeed.ops.adam import FusedAdam
```
Comment (SeanNaren) [inline docs/source/advanced/advanced_gpu.rst:372]:
```suggestion
```

**Review by awaelchli — APPROVED** [review]

File: docs/source/advanced/advanced_gpu.rst (hunk 2 above)
Comment (SeanNaren) [reply docs/source/advanced/advanced_gpu.rst]: yup thank you!
